src/main/java/ru/jru/quest/karaganov/service/FilterService.java

```diff
@@ -0,0 +1,28 @@
+package ru.jru.quest.karaganov.service;
+
+import lombok.NoArgsConstructor;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import java.util.HashMap;
+import java.util.Map;
+
+@NoArgsConstructor
+public class FilterService {
+
+    public Map<String, String> sessionInfoMap(HttpServletRequest request){
+        HashMap<String, String> resultMap = new HashMap<>();
+        HttpSession session = request.getSession();
+        String servletPattern = "/gameServlet";
+        if (request.getParameter("stepID") == null){
+            servletPattern = servletPattern + "?stepID=1";
+        }
+        resultMap.put("servletPattern", servletPattern);
+        if (session.getAttribute("IP")==null){
```

Comment:
formatting issue: should space before and after `==`

Reply:
fixed